### PR TITLE
Add & remove

### PR DESCRIPTION
**Pull request by Chawki-Mbarki**
# To Do list: add & remove 🚀 

In this pull request, I have implemented the CRUD (create, read, update, delete) methods for the To Do list project. All the elements of the user interface are fully functional.🎉

## Changes

- Remove all hardcoded items from the tasks array.🗑️
- Create a new JavaScript file for the new functionality.📄
- Implement a function for adding a new task (add a new element to the array).➕
- Implement a function for deleting a task (remove an element from the array).➖
- Implement a function for editing task descriptions.✏️
- Set the property completed to false and the property index to the value of the new array length by default for new tasks.☑️
- Update all remaining items' indexes when deleting a task.🔄
- Save all changes to the To Do List in local storage.💾

## Feedback

I would appreciate any feedback or suggestions on how to improve this project.😊

### CODE REVIEW

File: src/scripts/modules/functions/removeTask.mjs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import orderTasks from "./orderTasks.mjs";
+
+export default function removeTask(tasks, index) {
+  tasks.splice(index, 1)
+  const orderedTasks = orderTasks(tasks);
+  console.log(orderedTasks)
+  return orderedTasks;
+}
```
Comment (AnselemOdims) [inline src/scripts/modules/functions/removeTask.mjs]: - [x] Good job working on the adding and editing, but there is no button provisioned for removing tasks from the list. Please kindly include the remove functionality.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/scripts/modules/functions/removeTask.mjs [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
-  tasks.splice(index, 1)
+  if(index === tasks.length-1) {
+    tasks.pop();
+  }else if (index === 0){
+    tasks.shift();
+  }else{
+    tasks.splice(index, 1);
+  }
+
   const orderedTasks = orderTasks(tasks);
```
Comment (Shedrack-Sunday) [inline src/scripts/modules/functions/removeTask.mjs]: - [ ] Due to changes made to your codebase, i'm unable to add new task, which means i could not confirm the `remove` button asked by the previous reviwer. 
- Kindly review the bug initiated in your code that led to this, the verify you and add and remove task successfully 👍 

## View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @chawkimbarki    👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done  extremely well so far in this project,  However, there's one adjustment  you need to make  to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  

  
# Required Changes  🌵 


- [x] You made some impressive implementations which I must say is a job well done. However, there's one change I need you to make In line with the requirements.
- The changes you made from the last review has made me not to be able to add nor remove from your task
- When i cloned your project, i noticed a loader issue. Kindly see to this, and contact me if you need help.

## Loader issues:
image
 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

- Kindly add a `features` and `Future Features` section on your README. This is very important to make your repo professional 👍 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @chawkimbarki , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**